src/analysis/retail/mage/frost/core/WintersChillEvent.tsx

```diff
@@ -0,0 +1,157 @@
+  }
+
+  hasOneSpellAndRayOfFrost(): boolean {
+    const shatteredSpenders = this.damageEvents.filter((d) =>
+      WINTERS_CHILL_SPENDERS.includes(d.ability.guid),
+    );
+    const rayHits = this.damageEvents.filter(
+      (d) => d.ability.guid === TALENTS.RAY_OF_FROST_TALENT.id,
+    );
+    return shatteredSpenders.length === 1 && rayHits.length > 2;
+  }
+
+  getShatterPerformance(): QualitativePerformance {
+    if (this.hasTwoShatteredSpells() || this.hasOneSpellAndRayOfFrost()) {
+      return QualitativePerformance.Good;
+    }
+    return QualitativePerformance.Fail;
+  }
+
+  getShatterPerformanceMessage(): string {
+    let message = '';
+    if (!this.hasTwoShatteredSpells() && !this.hasOneSpellAndRayOfFrost()) {
+      message = 'Stacks not used';
+    } else if (this.hasTwoShatteredSpells()) {
+      message = '2 spells shattered';
+    } else if (this.hasOneSpellAndRayOfFrost()) {
+      message = '1 spell and Ray of Frost shattered';
+    }
+    return message;
+  }
+
+  getPrecastPerformance(): QualitativePerformance {
+    let performance = QualitativePerformance.Fail;
+    if (this.precast && this.precastInDamageEvents()) {
+      performance = QualitativePerformance.Good;
+    } else if (!this.precast && this.precastIcicles === 4) {
+      performance = QualitativePerformance.Good;
+    } else if (this.precast && !this.precastInDamageEvents()) {
+      performance = QualitativePerformance.Fail;
+    } else if (!this.precast) {
+      performance = QualitativePerformance.Fail;
+    }
+    return performance;
+  }
+
+  getPrecastPerformanceMessage(): string {
+    let message = 'Unknown fail';
+    if (this.precast && this.precastInDamageEvents()) {
+      message = 'Precast shattered';
+    } else if (!this.precast && this.precastIcicles === 4) {
+      message = 'No precast on 4 icicles';
+    } else if (this.precast && !this.precastInDamageEvents()) {
+      message = 'Precast not shattered';
+    } else if (!this.precast) {
+      message = 'No precast';
+    }
+    return message;
+  }
+
+  getPerformance(): QualitativePerformance {
+    let performance = QualitativePerformance.Fail;
+    if (
+      this.getPrecastPerformance() === QualitativePerformance.Good &&
+      this.getShatterPerformance() === QualitativePerformance.Good
+    ) {
+      performance = QualitativePerformance.Good;
+    }
+    return performance;
+  }
+
+  precastInDamageEvents() {
+    const preCastDamage: DamageEvent | undefined =
+      this.precast && GetRelatedEvent(this.precast, 'SpellDamage');
+    return preCastDamage && this.damageEvents.includes(preCastDamage);
+  }
+
+  getPerformanceDetails(): JSX.Element {
+    const spellsArray: Spell[] = Object.values(SPELLS);
+    const precast: Spell | undefined = spellsArray.find(
+      (spell) => spell.id === this.precast?.ability.guid,
+    );
+    const flurry: Spell | undefined = spellsArray.find(
+      (spell) => spell.id === this.flurry?.ability.guid,
+    );
```

Comment:
1. as I'm working right now on single target only, flurry will be constant, but other spells could apply winters chill too. I'm changing the name to applier for a wider support and better semantics.
2. precast could not exist because the normalizer only looks for frostbolt and glacial spike in a certain amount of time before the buff apply. As for flurry (applier now) I think that the cast not existing is a defensive programming thing that was already there.

I was looking for something like `maybeGetSpell` but didn't found it, thanks for pointing it out 👍